dictdiffer/__init__.py

```diff
@@ -35,7 +35,7 @@
 
 
 def diff(first, second, node=None, ignore=None, path_limit=None, expand=False,
-         tolerance=EPSILON, dot_notation=True):
+         tolerance=EPSILON, absolute_tolerance=None, dot_notation=True):
```

Comment:
looking at the code below with the `rel_diff and abs_diff`, it looks like you could get into a situation where you always have a diff if both tolerances are set to `None`. What if we reuse tolerance and add a parameter called `tolerance_type='relative'` with the other option being `'absolute'` (or an ENUM). Then we can have a single comparison unless you can think of a use case for having both comparisons.

Reply:
If both tolerances are `None`, then only the strictly equal numbers will be considered as equal thanks to line 262: `if first == second`.
I prefer having both comparison since I want to use the absolute threshold to ignore very small numbers and compare only the relevant ones.
For the interface I was inspired by the `numpy.isclose()` function which I find easy to understand and to use.